kernel/kernel-api/src/main/java/io/delta/kernel/internal/DeltaErrors.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal;
+
+public final class DeltaErrors {
+
+    // TODO update to be user-facing exception with future exception framework
+    //  (see delta-io/delta#2231) & document in method docs as needed (Table::getSnapshotAtVersion)
+    public static RuntimeException nonReconstructableStateException(
```

Comment:
these can be default visibility, right? (i.e. not public, but package-private)?

That way, only members of io.delta.kernel.internal can access it? Else, it is publicly visible.

Reply:
Cannot since accessed from different packages since some stuff is nested under `kernel/kernel-api/src/main/java/io/delta/kernel/internal`. The entire class is in `internal/` so it should be fine though?